### PR TITLE
feat(core-react): add text field 

### PR DESCRIPTION
**Pull request by adamraider**

### CODE REVIEW

File: packages/core-react/src/components/TextField/index.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  disabled,
+  required,
+  value,
+  error,
+  compact,
+  label,
+  ...rest
+}) {
+  const wrapperClass = clsx('ray-text-field', {
+    'ray-text-field--has-value': value,
```
Comment (i8ramin) [inline packages/core-react/src/components/TextField/index.js]:
```suggestion
    'ray-text-field--has-value': !!value,
```

File: packages/core-react/src/components/TextField/index.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  placeholder,
+  disabled,
+  required,
+  value,
+  error,
+  compact,
+  label,
+  ...rest
+}) {
+  const wrapperClass = clsx('ray-text-field', {
```
Comment (i8ramin) [inline packages/core-react/src/components/TextField/index.js]: Should this also handle textarea?

    'ray-text-field': !textarea,
    'ray-text-area': textarea,

File: packages/core-react/src/components/TextField/index.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        placeholder={placeholder}
+        disabled={disabled}
+        {...rest}
+      />
+      {label && (
+        <label className="ray-text-field__label" htmlFor={id}>
+          {label}
+        </label>
+      )}
+    </div>
```
Comment (cmugla) [inline packages/core-react/src/components/TextField/index.js]: should we also include the error message?

Comment (adamraider) [reply packages/core-react/src/components/TextField/index.js]: i was considering it... this pr is still wip

File: packages/core-react/src/components/TextField/index.js (hunk 2 above)
Comment (adamraider) [reply packages/core-react/src/components/TextField/index.js]: maaybeee.

File: packages/core-react/src/components/TextField/index.js (hunk 1 above)
Comment (adamraider) [reply packages/core-react/src/components/TextField/index.js]: why ?

**Pull request merged**